### ISSUE TITLE
Group DM's not showing

### ISSUE THREAD

**Issue by chainria** [EionRobb/purple-discord#275]
Today, after a long time, I joined a Group DM, created it on the official client.

Pidgin seems to be totally oblivious to it. I could fish it out of the chat room list, but:

* the whole history is missing
* the group DM does not appear on the contact list

Please tell me what I am doing wrong or how I can be of help to troubleshoot this problem!

**Comment by EionRobb** [comment]
I'm not sure about the 'auto add to buddy list' system, but you can manually add it to the buddy list either with the 'Add' button in the Tools->Room List window, or from the Conversations->Add shortcut when you're in the group chat

**Comment by chainria** [comment]
Yeah I have enabled the auto-add option, but somehow I think it does not work properly on Group DM's. Also, I guess history-handling is not implemented yet, if I read the code correctly. I wrote up a quick litte bash-script with some Python to retrieve the history myself, so that's no big problem for now.

So I guess the only thing remaining is that Group DM's are missed entirely by Pidgin using HEAD. Even now that I have the chat on my buddy list, incoming messages will simply be ignored - except if I set it to auto-join on login so the window stays open. I think there's something still not 100% right. Should you be interested, I generated a debug log.

**Comment by EionRobb** [comment]
What value do you have set for the "Number of users in a large channel" setting in the Advanced tab, and how many people are in the Group DM?

**Comment by chainria** [comment]
20, but only three people are in the Group DM

**Comment by EionRobb** [comment]
Hmm that's weird.  Just tested and for me the group DM window auto-opens, and messages display, even if its not on the buddy list.

Maybe I'll make that into an explicit setting for "always open group DM's", and ignore the 'large room size' setting.

A debug log would be really helpful, if you could capture one as you receive a group DM (in a perfect world, one log when it works and one log when it doesn't)

**Comment by chainria** [comment]
I cannot get it to open a group DM for the sake of god. Almost starting to suspect some other setting in Pidgin itself. I can even see the message in the debug log. But it just won't open.

**Comment by chainria** [comment]
Sorry, it took quite a while for me to debug this, which also had to do with my private load in life.

I can now positively say that two channels are neither bolded nor logged nor do they open up a new window. I have set the number of big channels to 100 which should be sufficient. Still, those two are not behaving like the rest does. For me this is quite a big problem as I try to get a useful log out of Discord this way. For now I set those channels to auto-join in Pidgin, what helps quite a lot because as soon as the window is opened, it will start logging like it should.

I took a deep look in the debug logs, but I couldn't find anything off. I also looked through the source multiple times and I just cannot figure out why this happens. And yes, even using a completely new folder reproduces the same problems. I am honestly out of ideas.

**Comment by baltitenger** [comment]
I'm having the same issue.

**Comment by chainria** [comment]
I found out this only happens when you create the room yourself. If you get invited, it indeed works. Maybe this helps?

**Comment by alexolog** [comment]
When I got invited to a group DM, it promptly crashed Pidgin.  #317

**Comment by EionRobb** [comment]
> I found out this only happens when you create the room yourself. If you get invited, it indeed works. Maybe this helps?

Actually, that helps heaps @chainria ! - there's a different event that gets fired when that happens!  If you have the debug window open (Help->Debug Window) while you create one, there should be an error message about "Unknown message ..." that needs handling

**Comment by chainria** [comment]
> > I found out this only happens when you create the room yourself. If you get invited, it indeed works. Maybe this helps?
> 
> Actually, that helps heaps @chainria ! - there's a different event that gets fired when that happens! If you have the debug window open (Help->Debug Window) while you create one, there should be an error message about "Unknown message ..." that needs handling

Unfortunately, it is not that simple (would have been to great!)

(08:44:38) discord: got frame data: {"t":"CHANNEL_CREATE","s":31813,"op":0,"d":{"type":3,"recipients":[{"username":"THEM","public_flags":0,"id":"641489012204240897","discriminator":"0001","avatar":null},{"username":"THEM","public_flags":256,"id":"320260353659240459","discriminator":"0002","avatar":"3c47c2998f6570fc4372b621a8c903b0"}],"owner_id":"131356438432972802","name":null,"last_message_id":null,"id":"783961881575555083","icon":null}}
...
(08:44:41) discord: got frame data: {"t":"MESSAGE_CREATE","s":31815,"op":0,"d":{"type":0,"tts":false,"timestamp":"2020-12-03T07:44:41.455000+00:00","referenced_message":null,"pinned":false,"nonce":"783961896368734208","mentions":[],"mention_roles":[],"mention_everyone":false,"id":"783961896592080899","flags":0,"embeds":[],"edited_timestamp":null,"content":".","channel_id":"783961881575555083","author":{"username":"ME","public_flags":128,"id":"131356438432972802","discriminator":"0000","avatar":"346038aa67e09a2cb075919722b7d664"},"attachments":[]}}

...and nothing happened. No new channel on the list, no new message popped up. But maybe you can see something out the ordinary? Because I sure don't. For privacy issues I changed the users in the log to ME#0000, THEM#0001 and THEM#0002 and filtered out things that didn't belong. Other than that, nothing to see unfortunately.
